### PR TITLE
add validation for icons that are too large

### PR DESCRIPTION
**Pull request by Pentarctagon**

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
I think "too large" is a bit ambiguous. I bet there will be someone who sees that message and thinks "Huh? It's only 72x72, how is that too large?". It should be made more clear that the issue is with respect to file size, not dimensions.

**Comment by Pentarctagon** [comment]
Updated.

**Comment by CelticMinstrel** [comment]
It's a bit of an awkward phrasing, but it's better than before. Might be nice to add the actual filesize and the limit to the message, but I guess it's fine like this. Inserting a filesize into a message probably requires adding something like a `format_memory` function anyway.

File: src/addon/validation.cpp [hunk 1]
```diff
@@ -501,6 +505,10 @@ std::string addon_check_status_desc(unsigned int code)
 			ADDON_CHECK_STATUS::AUTH_TYPE_MISMATCH,
 			N_("The add-on’s forum_auth attribute does not match what was previously uploaded.")
 		},
+		{
+			ADDON_CHECK_STATUS::ICON_TOO_LARGE,
+			N_("The addon’s icon’s file size is too large.")
```
Comment (Wedge009) [inline src/addon/validation.cpp:510]: I just note that the dominant spelling for 'add-on' in Wesnoth is hyphenated. (You already used this in `client.cpp` so not sure why it's inconsistent here. It's even different from the line above in this file.)

Comment (Pentarctagon) [reply src/addon/validation.cpp]: Updated.

Comment (Wedge009) [reply src/addon/validation.cpp]: Thanks.

**Comment by Vultraz** [comment]
I'm a bit confused how this check works. It's not actually checking file size, it's checking the string length of the path?

**Comment by knyghtmare** [comment]
> I'm a bit confused how this check works. It's not actually checking file size, it's checking the string length of the path?

This is for URI specifically, I guess. Since the URI is a string of big length

**Comment by Pentarctagon** [comment]
Right, since the add-ons server doesn't send an image file to clients requesting the add-ons list, it sends either a path string to a mainline image (very small) or a DataURI (can be too large).

**Comment by soliton-** [comment]
I do not understand https://github.com/wesnoth/wesnoth/pull/9667/commits/d7215a9bfb0e7486f6238a5a12b2893b9bac8e5e at all. With the move to a fixed size check this seems almost useless.

I'd remove the client side check and let the server decide.

**Comment by Pentarctagon** [comment]
Isn't the point of the various client side checks to let the person uploading know about issues before it tries to upload?

**Comment by soliton-** [comment]
> Isn't the point of the various client side checks to let the person uploading know about issues before it tries to upload?

Sure and if there was an obvious check for an **error** (like check if the image is bigger than 72x72) then this would make sense. But checking for a fixed arbitrary (currently very high) value can at most be a warning IMO. An arbitrary size check needs to be configurable and the configuration would be server-side.

What could be done for a client-side check that blocks uploading would be to send the server-side configured value to the client on connect. Then the client knows the exact value that actually will be rejected and can error out before trying to upload.

**Comment by Pentarctagon** [comment]
Is there any situation where we would actually want to allow people to upload DataURIs that are larger than 500 KB?

**Comment by soliton-** [comment]
What is a size check in the editor for?

**Comment by Pentarctagon** [comment]
There's the pbl editor dialog that will automatically create a DataURI from an external image. It should be able to validate whether the DataURI it creates is too large to be actually uploaded.

**Comment by soliton-** [comment]
Why does that not check the size in pixels of the input image?

**Comment by Pentarctagon** [comment]
It could also do that, but the issue is with the file size not with the image dimensions

**Comment by soliton-** [comment]
There is a made up issue with an arbitrary file size. An actual limit could be an image size of 72x72.

**Comment by Pentarctagon** [comment]
Well, feel free to implement your preferred solution then.

**Pull request merged**